lib/src/metta/text.rs

```diff
@@ -242,4 +286,58 @@ mod tests {
         let mut parser = SExprParser::new("(a))");
         while let Some(_) = parser.parse(&Tokenizer::new()) {}
     }
+    
+    #[test]
+    fn test_parser_comments(){
+        println!("test_parser_comments");
+        let mut i = 0;
+        let program = " (a ; 4)
+                  5)
+                !(match &self (a $W)   $W)";
+        let expected = vec!("(a 5)", "!", "(match &self (a $W) $W)");
+        println!("running test {}", i); i+=1;
+        parse_compare(program, &expected);
```

Comment:
Few comments here:
1. Each test should be as simple as it is possible while demonstrating the feature works. Thus `!(match &self (a $W) $W)` is not needed at first case for instance
2. Each test case ideally should be in separate test
3. Using custom results comparison procedure like `parse_compare` makes tests results less clear, in cases below using `assert_eq!` is enough 
4. Standard syntax for `vec!` macros is `vec![]` not `vec!()`

```rust
    fn parse(text: &str) -> Vec<Atom> {
        let tokenizer = Tokenizer::new();
        let mut parser = SExprParser::new(program);
        let mut result = Vec::new();
        while let Some(atom) = parser.parse(&tokenizer) {
            result.push(atom);
        }
        result
    }

    #[test]
    fn test_comment_out_tail_of_expression(){
        let atoms = parse("(a ; 4)
                  5)");
        assert_eq!(atoms, vec![expr!("a", "5")]);
    }

    #[test]
    fn test_comment_out_next_expression(){
        let atoms = parse("a ;;(a  1) b");
        parse_compare(atoms, vec![sym!("a"), sym!("b")]);
    }

    #[test]
    fn test_comment_out_multiline_expression(){
        let atoms = parse(";; (a
                        1)");
        assert_eq!(atoms, vec![]);
    }

    #[test]
    fn test_comment_out_empty_line(){
        let atoms = parse("(a 1);
            b
         ");
        assert_eq!(atoms, vec![expr!("a", "1"), sym!("b")]);
    }
```